.github/workflows/desktop.yml

```diff
@@ -3,54 +3,105 @@ name: Desktop Builds
 on:
   pull_request:
     types: [opened, reopened, synchronize]
+  workflow_dispatch:
+    inputs:
+      expanded_matrix:
+        description: 'Use an expanded test matrix?'
+        default: '0'
+        required: true
 
 env:
   CCACHE_DIR: ${{ github.workspace }}/ccache_dir
-  # Xcode version 11.7 is the version we build the SDK with.
-  # Our MacOS runners will use the version in /Applications/Xcode_${xcodeVersion}.app
-  xcodeVersion: "11.7"
 
 jobs:
+  prepare_matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix_os: ${{ steps.export-result.outputs.matrix_os }}
+      matrix_build_type: ${{ steps.export-result.outputs.matrix_build_type }}
+      matrix_architecture: ${{ steps.export-result.outputs.matrix_architecture }}
+      matrix_msvc_runtime: ${{ steps.export-result.outputs.msvc_runtime }}
+      matrix_vcpkg_triplet_suffix: ${{ steps.export-result.outputs.vcpkg_triplet_suffix }}
+      matrix_xcode_version: ${{ steps.export-result.outputs.xcode_version }}
+      matrix_python_version: ${{ steps.export-result.outputs.python_version }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: false
+      - name: Use expanded matrix
+        if: github.event.inputs.expanded_matrix == '1'
+        run: |
+          echo "EXPANDED_MATRIX_PARAM="-e=1" >> $GITHUB_ENV
+      - id: export-result
+        run: |
+          echo "::set-output name=matrix_os::$( python scripts/gha/print_matrix_configuration.py -w desktop -k os ${EXPANDED_MATRIX_PARAM})"
+          echo "::set-output name=matrix_build_type::$( python scripts/gha/print_matrix_configuration.py -w desktop -k build_type ${EXPANDED_MATRIX_PARAM})"
+          echo "::set-output name=matrix_architecture::$( python scripts/gha/print_matrix_configuration.py -w desktop -k architecture ${EXPANDED_MATRIX_PARAM})"
+          echo "::set-output name=msvc_runtime::$( python scripts/gha/print_matrix_configuration.py -w desktop -k msvc_runtime ${EXPANDED_MATRIX_PARAM})"
+          echo "::set-output name=vcpkg_triplet_suffix::$( python scripts/gha/print_matrix_configuration.py -w desktop -k vcpkg_triplet_suffix ${EXPANDED_MATRIX_PARAM})"
```

Comment:
I think vcpkg_triplet_suffix shouldn't be in the matrix parameters. This can be easily derived from other values in the matrix and avoiding this in the matrix will trim down the amount of "excludes" in the workflow.

Reply:
It could just be an environment variable that we define and pass to the steps.

Reply:
In the previous version of this workflow th VCKPKG suffix was being added to the matrix with include statements below. Unfortunatley the use of include restricts us from dynamically excluding some builds through the configuration above since the include would simply re-add the build to the job matrix again.

that is to say that the previous implementation meant that we had to run with all windows, linux and mac, always.

I'll see if I can change this in a timely manner, but since the original script also had vcpkg in the matrix this might be another story/ticket to resolve later.

Reply:
Fixed!